imageproxy.go

```diff
@@ -409,6 +410,14 @@ func (t *TransformingTransport) RoundTrip(req *http.Request) (*http.Response, er
 
 	defer resp.Body.Close()
 
+	if resp.StatusCode >= 400 {
+		if resp.StatusCode == 400 {
+			return nil, errors.ErrCustom.New("Bad Request (400) received when retrieving image.")
+		}
+		return nil,
+			errors.ErrCustom.New(fmt.Sprintf("Error received when retrieving image. Status: %d", resp.StatusCode))
```

Comment:
Similar to the above, we should include the URL in the error message. Additionally, we could set the error code in the error struct, using [`error.SetCode(resp.StatusCode)`](https://github.com/TotalWineLabs/go-common/blob/8eefabd660eddb84856ea464d3445bc980fe1150/src/errors/errors.go#L67) this should make the logged error more consistent with other services that are using `go-common/errors`.

Something like:
```go
error := errors.ErrCustom.New(fmt.Sprintf("Error received when retrieving image: %s",  u.String())
error.SetCode(resp.StatusCode)
return nil, error
```